src/lua/datetime.lua

```diff
@@ -430,6 +753,19 @@ local function datetime_now()
     return d
 end
 
+-- addition or subtraction from date/time of a given interval
+-- described via table direction should be +1 or -1
+local function datetime_shift(self, o, direction)
+    assert(direction == -1 or direction == 1)
+    local title = direction > 0 and "datetime.add" or "datetime.sub"
+    if type(o) ~= 'table' then
```

Comment:
The same problem (though slightly reverse) - argument may be not only table, but interval object.

Reply:
(See updated patch above)